applications/asset_tracker_v2/tests/lwm2m_integration/src/lwm2m_integration_test.c

```diff
@@ -0,0 +1,344 @@
+{
+	TEST_ASSERT_EQUAL(CLOUD_WRAP_EVT_DISCONNECTED, evt->type);
+}
+
+/* Tests */
+void test_lwm2m_integration_connect(void)
+{
+	uint32_t current_lifetime_expected = 0;
+	uint32_t new_lifetime_expected = CONFIG_LWM2M_ENGINE_DEFAULT_LIFETIME;
+
+	__wrap_lwm2m_rd_client_start_AddCallback(&rd_client_set_callback_stub);
+
+	/* After the uut has been put into state CONNECTED, the lwm2m lifetime resource is
+	 * updated. This is done by getting the resource that contains the lifetime and setting it
+	 * to the configured value if they do not match. This needs to be done when using
+	 * bootstrapping because the boostrap server will override the default value set by the
+	 * application.
+	 */
+	__wrap_lwm2m_engine_get_u32_ExpectAndReturn(LIFETIME_PATH, &current_lifetime_expected, 0);
+	__wrap_lwm2m_engine_set_u32_ExpectAndReturn(LIFETIME_PATH, new_lifetime_expected, 0);
+
+	__wrap_lwm2m_security_needs_bootstrap_ExpectAndReturn(0);
+	__wrap_lwm2m_rd_client_start_ExpectAndReturn(&client,
+						     endpoint_name,
+						     0,
+						     NULL,
+						     NULL,
+						     0);
+	__wrap_lwm2m_rd_client_start_IgnoreArg_event_cb();
+
+	TEST_ASSERT_EQUAL(cloud_wrap_connect(), 0);
+
+	/* Trigger internal events that puts the uut in CONNECTED state. */
+	rd_client_callback(&client, LWM2M_RD_CLIENT_EVENT_BOOTSTRAP_REG_COMPLETE);
+	rd_client_callback(&client, LWM2M_RD_CLIENT_EVENT_REGISTRATION_COMPLETE);
+}
+
+void test_lwm2m_integration_disconnect(void)
+{
+	test_lwm2m_integration_connect();
+
+	__wrap_lwm2m_rd_client_stop_ExpectAndReturn(&client, NULL, false, 0);
+	__wrap_lwm2m_rd_client_stop_IgnoreArg_event_cb();
+
+	TEST_ASSERT_EQUAL(0, cloud_wrap_disconnect());
+}
+
+void test_lwm2m_integration_data_send(void)
+{
+	/* Populate path with random resource path references. */
+	char *paths[PATH_LEN] = {
+		"4/0/6",
+		"4/0/7",
+		"4/0/7",
+		"4/0/7",
+		"4/0/7",
+	};
+
+	__wrap_lwm2m_engine_send_ExpectAndReturn(&client, (const char **)paths, PATH_LEN, true, 0);
+
+	TEST_ASSERT_EQUAL(0, cloud_wrap_data_send(NULL, PATH_LEN, true, 0, paths));
+}
+
+void test_lwm2m_integration_ui_send(void)
+{
+	/* Populate path with random resource path references. */
+	char *paths[PATH_LEN] = {
+		"4/0/6",
+		"4/0/7",
+		"4/0/7",
+		"4/0/7",
+		"4/0/7",
+	};
+
+	__wrap_lwm2m_engine_send_ExpectAndReturn(&client, (const char **)paths, PATH_LEN, true, 0);
+
+	TEST_ASSERT_EQUAL(0, cloud_wrap_ui_send(NULL, PATH_LEN, true, 0, paths));
+}
+
+void test_lwm2m_integration_neighbor_cells_send(void)
+{
+	/* Populate path with random resource path references. */
+	char *paths[PATH_LEN] = {
+		"4/0/6",
+		"4/0/7",
+		"4/0/7",
+		"4/0/7",
+		"4/0/7",
+	};
+
+	__wrap_lwm2m_engine_send_ExpectAndReturn(&client, (const char **)paths, PATH_LEN, true, 0);
+
+	TEST_ASSERT_EQUAL(0, cloud_wrap_neighbor_cells_send(NULL, PATH_LEN, true, 0, paths));
+}
+
+void test_lwm2m_integration_agps_request_send(void)
+{
+	/* Populate path with random resource path references. */
+	char *paths[PATH_LEN] = {
```

Comment:
Can you add a `const` qualifier to the char? That way you can avoid typecasting in line 254.

Reply:
arg, that slipped my mind. Thanks!

Reply:
I don't get it. But I will let it be. This need not be a blocker